server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -900,7 +900,10 @@ private boolean shouldUpgradePendingSegment(
     } else if (pendingSegment.getId().getVersion().compareTo(replaceVersion) >= 0) {
       return false;
     } else if (!replaceInterval.contains(pendingSegment.getId().getInterval())) {
-      return false;
+      throw new ISE(
+          "Incompatible interval[%s] for committing pending segment[%s].",
```

Comment:
```suggestion
          "Cannot upgrade pending segment[%s] to version[%s] as replace interval[%s] does not fully contain the pending segment interval[%s].",
```

While we are at it, could you also update the corresponding exception message in `createNewIdsOfAppendSegmentsAfterReplace`.
Also, please throw a `DruidException` of category unsupported instead of an ISE.

Reply:
Done.